### ISSUE TITLE
dictionary literal cannot be inferred as TypedDict generic parameter

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#6052]
```python
from dataclasses import dataclass
from typing import Generic, TypedDict, TypeVar

P = TypeVar("P", bound="Payload")


class Payload(TypedDict):
    source_id: str


class ArticlePayload(Payload):
    url_name: str


@dataclass
class Chunk(Generic[P]):
    payload: P


@dataclass
class Embedded(Generic[P]):
    chunks: list[Chunk[P]]


e = Embedded[ArticlePayload](
    chunks=[
        Chunk(
            # error: Argument of type "dict[str, str]" cannot be assigned to parameter "payload" of type "P@Chunk" in function "__init__"
            #     "url_name" is an undefined field in type "Payload" (reportGeneralTypeIssues)
            payload={"source_id": "abc", "url_name": "http://localhost"},
        )
    ],
)

```

The `payload` dictionary literal is inferred as `dict[str, str]` rather than `ArticlePayload` and therefore errors as above.

NB, given:
```python
            payload={"source_id": "abc"},
```
Then `payload` is inferred as `Payload`.

pyright 1.1.329
python 3.11.4

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.330, which I just published. It will also be included in an upcoming release of pylance.
